From 96f4e12153046a0f5efce6e13ec43865d985cece Mon Sep 17 00:00:00 2001
From: Uri Mashiach <uri.mashiach@compulab.co.il>
Date: Sun, 24 Jan 2021 15:58:13 +0200
Subject: [PATCH 62/89] iot-gate-imx8: add support for the IE-CL420 module

The IE-CL420 implements one channel ADC.

The CAN, TPM and the CL420 modules are 3 alternatives in the baseboard
IED-BASE, slots A and B for the SPI bus.

Add 2 disabled ADC nodes for the IE-CL420 modules.
The appropriate nodes will be enabled by the U-Boot, according to the
detected modules.

Signed-off-by: Uri Mashiach <uri.mashiach@compulab.co.il>
---
 .../boot/dts/compulab/sb-iotgimx8-ied.dts     | 22 +++++++++++++++++++
 arch/arm64/configs/iot-gate-imx8_defconfig    |  1 +
 2 files changed, 23 insertions(+)

diff --git a/arch/arm64/boot/dts/compulab/sb-iotgimx8-ied.dts b/arch/arm64/boot/dts/compulab/sb-iotgimx8-ied.dts
index 5e92ac790159..22f93994ac05 100644
--- a/arch/arm64/boot/dts/compulab/sb-iotgimx8-ied.dts
+++ b/arch/arm64/boot/dts/compulab/sb-iotgimx8-ied.dts
@@ -67,6 +67,14 @@
 		regulator-always-on;
 		enable-active-high;
 	};
+
+	vref_adc: regulator-vref-adc {
+		compatible = "regulator-fixed";
+		regulator-name = "vref_adc";
+		regulator-min-microvolt = <2400000>;
+		regulator-max-microvolt = <2400000>;
+		regulator-always-on;
+	};
 };
 
 &i2c4 {
@@ -129,6 +137,13 @@
 		spi-max-frequency = <5000000>;
 		status = "disabled";
 	};
+	adc0: adc@0 {
+		compatible = "maxim,max11108";
+		reg = <0>;
+		vref-supply = <&vref_adc>;
+		spi-max-frequency = <20000000>;
+		status = "disabled";
+	};
 };
 
 &ecspi3 {
@@ -159,6 +174,13 @@
 		spi-max-frequency = <5000000>;
 		status = "disabled";
 	};
+	adc1: adc@0 {
+		compatible = "maxim,max11108";
+		reg = <0>;
+		vref-supply = <&vref_adc>;
+		spi-max-frequency = <20000000>;
+		status = "disabled";
+	};
 };
 
 &iomuxc {
diff --git a/arch/arm64/configs/iot-gate-imx8_defconfig b/arch/arm64/configs/iot-gate-imx8_defconfig
index f80375e87389..dda66fc3f5d9 100644
--- a/arch/arm64/configs/iot-gate-imx8_defconfig
+++ b/arch/arm64/configs/iot-gate-imx8_defconfig
@@ -559,6 +559,7 @@ CONFIG_EXTCON_USB_GPIO=y
 CONFIG_MEMORY=y
 CONFIG_IIO=y
 CONFIG_IMX8QXP_ADC=y
+CONFIG_MAX11108=m
 CONFIG_PWM=y
 CONFIG_PWM_IMX27=y
 CONFIG_RESET_IMX7=y
-- 
2.17.1

